internal/commands/push/command.go

```diff
@@ -100,16 +100,17 @@ func (cmd *Command) Handler(profile *user.Profile, ui terminal.UI, clients cli.C
 		return err
 	}
 
-	var uploadPathDependencies string
+	var uploadDependenciesPath string
 	var dependenciesDiffs realm.DependenciesDiff
 	if cmd.inputs.IncludeDependencies {
-		uploadPathDependencies, err = local.PrepareDependencies(app, ui)
+		uploadDependencies, err := local.FindAppDependencies(app.RootDir)
 		if err != nil {
 			return err
 		}
-		defer os.Remove(uploadPathDependencies) //nolint:errcheck
+		uploadDependenciesPath := uploadDependencies.ArchivePath
+		defer os.Remove(uploadDependenciesPath) //nolint:errcheck
```

Comment:
I may be missing something still with the new way we're handling dependencies, but I don't think we want/need this `defer` anymore.  these `os.Remove` calls were always to ensure the CLI cleaned up after itself while creating new files from the `PrepareDependencies` implementation